jwplatform/client.py

```diff
@@ -219,28 +240,181 @@ def clip(self, site_id, channel_id, event_id, body=None, query_params=None):
+            upload_context: The query parameters.
+            **kwargs: The upload parameters.
+
+        Returns: None
+        """
+        if not upload_context:
+            raise ValueError("The provided context is None. Cannot resume the upload.")
+        if not upload_context.can_resume():
+            upload_context = self.create_media_and_get_upload_context(file, **kwargs)
+        upload_handler = self._get_upload_handler_for_upload_type(upload_context, file, **kwargs)
+        try:
+            upload_handler.upload()
+        except Exception:
+            file.seek(0, 0)
+            raise
+
+
+    def _get_upload_handler_for_upload_type(self, context: UploadContext, file, **kwargs):
+        upload_method = context.upload_method
+        base_url = kwargs.get('base_url', UPLOAD_BASE_URL)
+        target_part_size = int(kwargs.get('target_part_size', MIN_PART_SIZE))
+        retry_count = int(kwargs.get('retry_count', RETRY_COUNT))
+
+        if upload_method == UploadType.direct.value:
+            direct_link = context.direct_link
+            upload_handler = SingleUpload(direct_link, file, retry_count, context)
+        else:
+            upload_token = context.upload_token
+            upload_client = _UploadClient(api_secret=upload_token, base_url=base_url)
+            upload_handler = MultipartUpload(upload_client, file, target_part_size,
+                                             retry_count, context)
+        return upload_handler
+
+
+class _UploadClient(_ScopedClient):
+    _collection_path = "/v1/uploads/{resource_id}"
+
+    def __init__(self, api_secret, base_url):
+        if base_url is None:
+            base_url = UPLOAD_BASE_URL
+        client = JWPlatformClient(secret=api_secret, host=base_url)
+        super().__init__(client)
+
+    def list(self, upload_id, query_params=None):
+        """
+        Lists the parts for a given multi-part upload.
+        Args:
+            upload_id: The upload ID for the upload
+            query_params: The query parameters.
+
+        Returns: None
+
+        """
+        resource_path = self._collection_path.format(resource_id=upload_id)
+        resource_path = f"{resource_path}/parts"
+        response = self._client.request_with_retry(method="GET", path=resource_path, query_params=query_params)
+        return ResourcesResponse.from_client(response, 'parts', self.__class__)
+
+    def complete(self, upload_id, body=None) -> None:
+        """
+        Marks the upload as complete.
+        Args:
+            upload_id: The upload ID for the upload
+            body: [Optional] - The body of the payload.
+
+        Returns: List of parts with their upload metadata in a JSON format.
+
+        """
+        resource_path = self._collection_path.format(resource_id=upload_id)
+        resource_path = f"{resource_path}/complete"
+        self._client.request_with_retry(method="PUT", path=resource_path, body=body)
+        return
```

Comment:
```suggestion
        return self._client.request_with_retry(method="PUT", path=resource_path, body=body)
```

Reply:
@saurabhc123 I'm sorry I thought it was my PR